protos/flyteidl/core/workflow.proto

```diff
@@ -68,6 +68,32 @@ message WorkflowNode {
     }
 }
 
+// SignalCondition represents a dependency on an existing signal.
+message SignalCondition {
+    // A unique identifier for the requested signal.
+    string signal_id = 1;
+
+    // A type denoting the required value type for this signal.
+    LiteralType type = 2;
```

Comment:
Are there broader docs for how this is used? I didn't see an issue as part of the PR and I don't think I quite follow how this eventual value is produced. Could there be a case where we want multiple values as part of a signal?

Reply:
The best resource is probably the [RFC](https://github.com/flyteorg/flyte/pull/2090). Currently the implementation is setup so FlytePropeller calls the GetOrCreate RPC endpoint while evaluating the GateNode during each iteration. The node only succeeds if the signal has a value set.

An external entity will set the signal value. For example, a user through the SetSignal RPC endpoint or the UI can set the signal with a post to the RESTful interface.

And I believe the LiteralType proto encapsulates a Collection, Map, etc. So we should be able to support a variety of variable types, including multiple values.